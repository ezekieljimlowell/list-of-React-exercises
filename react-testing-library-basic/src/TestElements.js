import React, { useState } from "react";

const TestElements = () => {
  const [counter, setCounter] = useState(0);

  return (
    <>
      <h1 data-testid="counter">Counter is {counter}</h1>
      <button
        type="button"
        data-testid="button-up"
        onClick={() => setCounter(counter + 1)}
      >
        Increment
      </button>
      <button
        type="button"
        data-testid="button-down"
        onClick={() => setCounter(counter - 1)}
      >
        Decrement
      </button>
    </>
  );
};

export default TestElements;
