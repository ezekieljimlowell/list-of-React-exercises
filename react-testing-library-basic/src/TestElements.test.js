import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TestElements from "./TestElements";

it("should equal to zero", () => {
  const { getByText } = render(<TestElements />);
  expect(getByText("Counter is 0")).toBeInTheDocument();
});

it('should be enabled', () => {
  const { getByTestId } = render(<TestElements />);
  expect(getByTestId("button-up")).not.toHaveAttribute("disabled");
})

it('should be disabled', () => {
    const { getByTestId } = render(<TestElements />);
    expect(getByTestId("button-down")).not.toBeDisabled();
})

it("increment", () => {
    const { getByTestId } = render(<TestElements />);
    fireEvent.click(getByTestId("button-up"));
    expect(getByTestId("counter")).toHaveTextContent(1);
})

it("decrement", () => {
    const { getByTestId } = render(<TestElements />);
    fireEvent.click(getByTestId("button-down"));
    expect(getByTestId("counter")).toHaveTextContent(-1);
})

it("should take a snapshot", () => {
    const { asFragment } = render(<TestElements />);
    expect(asFragment(<TestElements />)).toMatchSnapshot();
  });
