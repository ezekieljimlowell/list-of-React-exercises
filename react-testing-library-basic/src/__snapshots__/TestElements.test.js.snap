// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <h1
    data-testid="counter"
  >
    Counter is 0
  </h1>
  <button
    data-testid="button-up"
    type="button"
  >
    Increment
  </button>
  <button
    data-testid="button-down"
    type="button"
  >
    Decrement
  </button>
</DocumentFragment>
`;
